llarp/messages/link_message_parser.cpp

```diff
@@ -0,0 +1,150 @@
+#include <messages/link_message_parser.hpp>
+
+#include <messages/dht_immediate.hpp>
+#include <messages/discard.hpp>
+#include <messages/link_intro.hpp>
+#include <messages/link_message.hpp>
+#include <messages/relay_commit.hpp>
+#include <messages/relay.hpp>
+#include <router_contact.hpp>
+#include <util/buffer.hpp>
+#include <util/logger.hpp>
+#include <util/metrics.hpp>
+
+#include <memory>
+
+namespace llarp
+{
+  struct InboundMessageParser::msg_holder_t
```

Comment:
given the nature of this PR, perhaps could/should also rename this to LinkMessageParser?

Reply:
Will do.